Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-dashboard/add-category-dialog.component.ts

```diff
@@ -1,8 +1,63 @@
 ﻿import { Component } from '@angular/core';
+import { Category } from '../category.model';
+import { CategoryService } from '../categories.service';
+import { MdDialogRef, MdSnackBar } from '@angular/material';
+import { QuestionsDashboardComponent } from './questions-dashboard.component'
 
 @Component({
     moduleId: module.id,
     selector: 'add-category-dialog',
     templateUrl: 'add-category-dialog.html'
 })
-export class AddCategoryDialogComponent { }
+
+export class AddCategoryDialogComponent {
+    private response: any;
+    private successMessage: string;
+
+    isCategoryNameExist: boolean;
+    errormessage: string;
+    category: Category;
+    responseobject: Category;
+
+    constructor(private categoryService: CategoryService, private dialogRef: MdDialogRef<AddCategoryDialogComponent>, public snackBar: MdSnackBar) {
+        this.isCategoryNameExist = false;
+        this.category = new Category();
+        this.successMessage = 'Category Name Added Sucessfully';
+    }
+
+    /**
+     * Open snackBar
+     */
+    openSnackBar(message: string) {
+        let snackBarRef = this.snackBar.open(message, '', {
+            duration: 3000,
+        });
+    }
+
+    /**
+     *Method to add Category 
+     * @param category:Category object
+     */
+    addCategory(category: Category) {
+        if (typeof category.categoryName !== 'undefined') {
+            this.categoryService.addCategory(category).subscribe(
+                result => {
+                    this.responseobject = result;
```

Comment:
responseObject not responseobject. Naming conventions are not being followed here.